pkg/scraper/client/resource/decode.go

```diff
@@ -51,6 +52,8 @@ func decodeBatch(b []byte, defaultTime time.Time, nodeName string) *storage.Metr
 		if et, err = parser.Next(); err != nil {
 			if err == io.EOF {
 				break
+			} else {
+				return nil, fmt.Errorf("failed parsing metrics: %w", err)
```

Comment:
Why not skip the parse error entry (only print error message) instead of returning error directly?
I think the data of one pod is abnormal  should not affect other pods to collect data

Reply:
can implement custom prometheus parser, but it is dangerous to trust following content after encountering unexpected token.
another point is that should not expect the data from other service is always ok to parse and stuck in an infinite loop.

Reply:
Agree with @AtkinsChang